cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -76,48 +75,59 @@
  * and annotations:
  *
  * cdap.twill.app=[literal app name]
+ *
+ * @param <T> type of the K8s resource
  */
-public class KubeTwillRunnerService implements TwillRunnerService {
-
-  static final String START_TIMEOUT_ANNOTATION = "cdap.app.start.timeout.millis";
-
+public class KubeTwillRunnerService<T> implements TwillRunnerService {
   private static final Logger LOG = LoggerFactory.getLogger(KubeTwillRunnerService.class);
 
+  static final String START_TIMEOUT_ANNOTATION = "cdap.app.start.timeout.millis";
   private static final String APP_ANNOTATION = "cdap.twill.app";
   private static final String APP_LABEL = "cdap.twill.app";
   private static final String RUNNER_LABEL = "cdap.twill.runner";
   private static final String RUNNER_LABEL_VAL = "k8s";
   private static final String RUN_ID_LABEL = "cdap.twill.run.id";
 
-  private static final String AVAILABLE_TYPE = "Available";
-
   private final String kubeNamespace;
   private final String resourcePrefix;
   private final Map<String, String> cConf;
   private final PodInfo podInfo;
   private final DiscoveryServiceClient discoveryServiceClient;
   private final Map<String, String> extraLabels;
-  private final DeploymentWatcher deploymentWatcher;
+  private final KubeResourceWatcher kubeResourceWatcher;
```

Comment:
Missing type parameter

Reply:
Since the class `KubeTwillRunnerService` is not parameterized and we are using same service for different resource types, I added 2 watcher variables one for deployment and another for statefulset explicitly. Let me know how it looks.